crates/ra_db/src/input.rs

```diff
@@ -108,11 +108,44 @@ struct CrateData {
     file_id: FileId,
     edition: Edition,
     dependencies: Vec<Dependency>,
+    cfg_options: CfgOptions,
+}
+
+#[derive(Debug, Clone, PartialEq, Eq, Default)]
+pub struct CfgOptions {
+    atoms: FxHashSet<SmolStr>,
+    features: FxHashSet<SmolStr>,
+    options: FxHashMap<SmolStr, SmolStr>,
+}
+
+impl CfgOptions {
+    pub fn check(&self, cfg: &CfgAttrKind) -> Option<bool> {
```

Comment:
Let's pro-actively move `Cfg` to a separate module, which contains three main things:

* the `CfgOptions` which define which cfgs are enabled
* `Cfg` expressions, which can be evaluated against options
* the evaluator